.github/workflows/beta.yml

```diff
@@ -31,69 +36,100 @@ jobs:
+          currentVersion=$(/home/runner/.cargo/bin/taplo get -f lib/Cargo.toml "package.version")
+          major=$(echo $currentVersion | tr "." "\n" | sed -n 1p)
+          minor=$(echo $currentVersion | tr "." "\n" | sed -n 2p)
+          patchAndMeta=$(echo $currentVersion | tr "." "\n" | sed -n 3p)
+          betaNum=$(echo $currentVersion | tr "." "\n" | sed -n 4p)
+          betaNum=$(($betaNum + 1))
+          version=${major}.${minor}.${betaNum}
+          betaVersion=${major}.${minor}.${patchAndMeta}.${betaNum}
+          
+          # Create the tag
+          git tag -a v${betaVersion} -m "Release ${betaVersion}"
 
-          # Unstash Cargo.toml
-          mv $tempDir/Cargo.toml lib/
+          # Bump the crate version
+          sed -i "s#^version = \".*\"#version = \"${betaVersion}\"#" Cargo.toml
+          sed -i "s#^version = \".*\"#version = \"${betaVersion}\"#" lib/Cargo.toml
 
-          # Replace the crate name
-          # We are just going to replace the first occurance of surrealdb
-          sed -i "0,/surrealdb/s//surrealdb-beta/" lib/Cargo.toml
+      - name: Commit changes
+        run: |
+          # Bump lock file without updating dependencies and ensure crate still compiles
+          cargo check --no-default-features --features storage-mem
 
           # Commit changes
           git checkout -b beta
-          git add lib/Cargo.toml
-          git commit -m "Prepare the beta crate"
-
-      - name: Perfom release checks
-        if: ${{ !inputs.publish }}
-        env:
-          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
-        run: /home/runner/.cargo/bin/release-plz release --dry-run --config .config/release-plz.toml
+          git commit -am "Prepare release"
+
+  release:
+    name: Prepare beta release
+    needs: [beta-branch]
+    uses: ./.github/workflows/reusable_publish_version.yml
+    with:
+      name: beta
+      environment: beta
+      git-ref: beta
+      publish: ${{ inputs.publish }}
+      create-release: ${{ inputs.publish }}
+    secrets: inherit
+
+  release-tag:
+    name: Push release tag
+    needs: [release]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          ref: beta
 
-      - name: Publish the crate
+      - name: Push the new branch and tag
         if: ${{ inputs.publish }}
-        env:
-          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
-          GIT_TOKEN: ${{ secrets.RELEASE_PLZ_TOKEN }}
         run: |
-          # Upload the crate
-          /home/runner/.cargo/bin/release-plz release --config .config/release-plz.toml
-
-          # Push the new branch and tag
-          git remote add upstream https://tobiemh:${{ secrets.RELEASE_PLZ_TOKEN }}@github.com/surrealdb/surrealdb.git
+          git remote add upstream https://tobiemh:${{ secrets.GITHUB_TOKEN }}@github.com/surrealdb/surrealdb.git
           git push upstream
           git push upstream --tags
```

Comment:
What do you intend to push here? Tags created in other jobs won't be available here.